src/utils/lstring.ts

```diff
@@ -2,24 +2,28 @@ import { Locales } from "../data/un-languages";
 import type { Locale } from "../types/lstring";
 import type LString from "../types/lstring";
 
-export default function (ltext: string | LString | { [locale: string]: string } | null | undefined, preferedLocale?: Locale): string {
+export default function (ltext?: string | LString | { [locale: string]: string }, ...preferedLocales: Locale[]): string {
   if (!ltext) return "";
 
   if (typeof (ltext) === 'string') return ltext;
 
-  const locale = preferedLocale || "en"; // TODO default to i18n.locale instead ;
+  const cleanedLtext: LString | { [locale: string]: string } = Object.entries(ltext)
+    .filter(isValidLocaleEntry)
+    .reduce((o, [k, v]) => ({ ...o, [k]: v }), {});
+  const ltextLocales = Object.keys(cleanedLtext);
+  const locale = preferedLocales.find((l) => ltextLocales.includes(l)) || "en"; // TODO default to i18n.locale instead ;
```

Comment:
it is a util. it needs to be stateless. it will never have i18n integrated to ti
```suggestion
  const locale = preferedLocales.find((l) => ltextLocales.includes(l)) || "en";
```